### DISCUSSION TITLE
IPS drop in Krypton Egg

### DISCUSSION THREAD

**Discussion by Vort** [bochs-emu/Bochs#192] [Q&A]
I made configuration with `ips=40000000` and `sync=slowdown`.
But when I launch Krypton Egg game, IPS suddenly drops 10 times, to 4000000:
![image](https://github.com/bochs-emu/Bochs/assets/1242858/ecd07fac-7586-4ae5-bc54-2fa1c4b2dcd6)

Can anyone say
1. What mechanism allows game to do this?
2. Why game may need to do this?

Because game lags horribly afterwards, I expect it's a bug.
But because this situation is very strange,  I decided to make discussion instead of issue.

Test files: [Krypton.zip](https://github.com/bochs-emu/Bochs/files/13761927/Krypton.zip).

**Comment by stlintel** [comment]
Does it write a lot of garbage to the log at the same time ?
Usually this is the reason for huge IPS drops

**Reply by Vort** [reply]
No.

**Comment by stlintel** [comment]
I can't reproduce it. The game runs with 40M as expected and never drops to 4M.

In the other hand sees like I got some sound initialization issue:
00213493263e[WAVEIN] Couldn't open wave device for recording (error = 2)
00213948668e[WAVEIN] Couldn't add buffer for recording (error = 34)
00214404073e[WAVEIN] Couldn't add buffer for recording (error = 34)
so may be the problem you see is sound emulation related.

BTW, the game lags horribly regardless of the IPS :)

**Reply by Vort** [reply]
> so may be the problem you see is sound emulation related.

You are right. No IPS drop with disabled `sb16`.
